src/stratigraphy/util/geometric_line_utilities.py

```diff
@@ -223,36 +229,106 @@ def merge_parallel_lines(lines: list[Line], tol: int = 8, angle_threshold: float
         return merged_lines
 
 
+def odr_regression(x: ArrayLike, y: ArrayLike) -> tuple[float, float]:
+    """Perform orthogonal distance regression on the given data.
+
+    Solves the line equation y = mx + c for the best fit values of m and c.
+    Different to linear regression, ODR takes into account the error in both x and y.
+    As such, it can handle horizontal and vertical lines.
+
+    Note: For vertical lines, the slope is nan.
```

Comment:
Are you sure about this? I tried to reproduce it, but I actually got a slope of `0` even for points that are on a vertical line. E.g.

```
    x = np.array([1, 1, 1])
    y = np.array([0, 1, 2])
    slope, intercept = odr_regression(x, y)
    print(slope, intercept)
```

Reply:
Let's write some unit tests for this function.